### PR TITLE
events: Add support for bulk delete_message events.

### PR DESCRIPTION
**Pull request by chrisbobbe**
I haven't yet tested end-to-end an actual bulk delete; I'm still looking for a way to do that in the UI on CZO. Maybe it's an admin feature, or it's not accessible through the UI yet?

Fixes: #4158

### CODE REVIEW

**Comment by chrisbobbe** [comment]
OK, just tested this on a dev server, and it works as expected. I see the `EVENT_MESSAGE_DELETE` action with multiple items in `messageIds`, and the messages do disappear.

File: src/message/__tests__/messagesReducer-test.js [hunk 1]
```diff
@@ -117,7 +129,7 @@ describe('messagesReducer', () => {
       });
       const action = deepFreeze({
         type: EVENT_UPDATE_MESSAGE,
-        messageId: 3,
+        messageId: [3],
```
Comment (gnprice) [inline src/message/__tests__/messagesReducer-test.js:132]: Hmm, it's surprising this one makes it past the type-checker! Would be good to identify why.

(I think this change is unintentional and can be simply dropped.)

File: src/events/eventToAction.js [hunk 2]
```diff
@@ -88,11 +88,15 @@ export default (state: GlobalState, event: $FlowFixMe): EventAction => {
...
+    // Before server feature level 13, or if we don't specify the
+    // `bulk_message_deletion` client capability (which we do) this
+    // event has `message_id` instead of `message_ids`.
+    case 'delete_message': {
       return {
         type: EVENT_MESSAGE_DELETE,
-        messageId: event.message_id,
+        messageIds: event.message_ids ?? [event.message_id],
       };
+    }
```
Comment (gnprice) [inline src/events/eventToAction.js:99]: nit: extra braces not needed

(I'm guessing they came from a previous draft of this commit where you perhaps had a local variable.)

File: src/message/__tests__/messagesReducer-test.js (hunk 1 above)
Comment (chrisbobbe) [reply src/message/__tests__/messagesReducer-test.js:132]: Thanks for catching this!

> Hmm, it's surprising this one makes it past the type-checker!

Alas, this file isn't type-checked at all.

File: src/events/eventToAction.js (hunk 2 above)
Comment (chrisbobbe) [reply src/events/eventToAction.js:99]: Yep, that was it—thanks for catching this.

**Comment by chrisbobbe** [comment]
Merged. Thanks for the review!

File: src/message/__tests__/messagesReducer-test.js (hunk 1 above)
Comment (gnprice) [reply src/message/__tests__/messagesReducer-test.js]: Ah, of course :-/ Looks like that's still true of a little over half our test files:
```
$ git ls src/**/__tests__/*.js | wc -l
92

$ git grep -l '@flow' src/**/__tests__/*.js | wc -l
39

**Pull request merged**